src/components/Results.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from "react";
+import TableHead from './TableHead';
+import TableBody from './TableBody';
+import * as FakeBookings from "../data/fakeBookings.json";
+
+
+class Results extends React.Component {
+	constructor(props) {
+		super(props)
+
+		this.state = {
+			data: [],
+			isFetched: false
+		}
+
+		this.checkIfFetched = this.checkIfFetched.bind(this);
+		this.getData = this.getData.bind(this);
+
+		}
+
+		checkIfFetched(event) {
+			console.log('checkIfFetched is here');
+			if (this.state.isFetched === false) {
+				this.getData();
+			}
+		}
+		
+
+		getData() {
+			console.log('this is', FakeBookings);
```

Comment:
As far as I can tell, there's no reason to split the `checkIfFetched` and `getData` methods?

Reply:
I separated them to make clear that handlers do different things - just check and fetch. But you are right, they can be in one